src/core/qgsofflineediting.cpp

```diff
@@ -164,157 +165,189 @@ void QgsOfflineEditing::synchronize()
   const QgsSnappingConfig snappingConfig = QgsProject::instance()->snappingConfig();
 
   // restore and sync remote layers
-  QList<QgsMapLayer *> offlineLayers;
   QMap<QString, QgsMapLayer *> mapLayers = QgsProject::instance()->mapLayers();
+  QMap<int, std::shared_ptr<QgsVectorLayer>> remoteLayersByOfflineId;
+  QMap<int, QgsVectorLayer *> offlineLayersByOfflineId;
+
   for ( QMap<QString, QgsMapLayer *>::iterator layer_it = mapLayers.begin() ; layer_it != mapLayers.end(); ++layer_it )
   {
-    QgsMapLayer *layer = layer_it.value();
+    QgsVectorLayer *offlineLayer( qobject_cast<QgsVectorLayer *>( layer_it.value() ) );
 
-    if ( layer->customProperty( CUSTOM_PROPERTY_IS_OFFLINE_EDITABLE, false ).toBool() )
+    if ( !offlineLayer || !offlineLayer->isValid() )
     {
-      if ( !layer->isValid() )
-      {
-        QgsDebugMsgLevel( QStringLiteral( "Skipping offline layer %1 because it is an invalid layer" ).arg( layer->id() ), 4 );
-        continue;
-      }
-
-      offlineLayers << layer;
+      QgsDebugMsgLevel( QStringLiteral( "Skipping offline layer %1 because it is an invalid layer" ).arg( offlineLayer->id() ), 4 );
```

Comment:
`if ! offlineLayer` -> `offlineLayer->id()` will crash, only debug the id if invalid

Reply:
I can also use the `layer_it.key()` here. Done.